easybuild/framework/easyconfig/format/version.py

```diff
@@ -499,7 +499,6 @@ def _get_all_toolchain_names(self, search_string=''):
         Initialise each search_toolchain request, save in module constant TOOLCHAIN_NAMES.
         :param search_string: passed to search_toolchain function.
         """
-        global TOOLCHAIN_NAMES
```

Comment:
This is where my ignorance shows, but don't we need this for line 505?

Reply:
There's only one `TOOLCHAIN_NAMES`, so the `global` statement is overkill.

Quick test without `global`:
```python
>>> TEST = {}
>>> class Alan(object):
...     def test(self):
...         TEST['one'] = 1
...
>>> a = Alan()
>>> TEST
{}
>>> a.test()
>>> TEST
{'one': 1}
```

Reply:
The point of "global" is that an assignment to the symbol affects the global symbol:
```
TEST = []
def foo():
    TEST.append(42)  # Access to existing symbol -> global
    TEST = [1,2,3]  # new local variable
    TEST.append(42)  # Access to local symbol shadowing global symbol
```

PyLint and others actually warn when shadowing a global name because this behavior is hard to reason about.